refactor: chat messages

saledjenic opened the pull request:
This PR introduces a struct (with as minimal as possible changes) on the qml side which is able to display messages from the refactored backend. We have a general feature now we are ready to build other message related features based on the changes done in this PR. The reason why we have a lot of commented out lines here is because of incompatibility between  a qml structure and a structure we have on the backend.

Ideally would be if we could redo complete qml to follow architecture we have on the backend, but till then we need to update the rest according to the minimal changes applied here.

Corresponding `status-lib` PR:
- https://github.com/status-im/status-lib/pull/128

What's done:
- backend part for `clean chat history` feature
- backend part for `mark messages as read` feature
- `chat service` is not `QtObject` inherited class any more
- mute/unmute methods added to `chat service`
- `type` added to sub items of model used in `chat section` module
- shared message item and model updated with set of new fields
- `chatType` added to `Constants` to group chat type related values
- `messageContentType` added to `Constants` to group message content type related values
- `MessageStore` maintained per message list now
- added new `ChatContentView` component used to display chat/channel content
- components updated to display messages from the new backend
- `StatusChatToolBar` appropriately updated
- mute/unmute feature added to chat/channel list as well as to `StatusChatToolBar`

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 6bbb402a | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-4265/1/) | 2021-12-07 14:32:10 | ~9 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-211207-142300-6bbb40-pr4265.dmg) |

jrainville commented on ui/imports/shared/views/chat/MessageView.qml:
```diff
@@ -265,21 +342,29 @@ Column {
+            color: "blue"
+            width: 100
+            height: 100
         }
```
Is this a leftover?

saledjenic replied:
@jrainville I left that intentionally since `ChannelIdentifierView` qml component is not refactored yet but `channelIdentifierComponent` is used in `messageLoader` and since `Component` cannot be empty I just commented `ChannelIdentifierView` and add that dummy rectangle in order to have compilable code. Will work on that component today.

jrainville reviewed (commented):
Good job. Very massive work

I mostly left optional change suggestions and some small questions

saledjenic commented on ui/app/AppLayouts/Chat/views/ChatContentView.qml:
```diff
@@ -0,0 +1,305 @@
+        Layout.fillWidth: true
+
+        chatInfoButton.title: chatContentModule.chatDetails.name
+        chatInfoButton.subTitle: chatContentRoot.determineSubTitle()
```
> does this update?

@jrainville good catch, I didn't check this. I updated the code and now `chatContentModule.chatDetails.type` will surely have effects to `subTitle`.

saledjenic commented:
@jrainville I updated all inline `if` conditions you mentioned, and I agree, maybe it's easier to read it if we add ` ` space after a condition, but that's more about coding style/preferences, so don't blame me if I write it again without ` ` space, cause I used to that "compact" mode of writing inline `if` a lot.  :)

jrainville reviewed (approved):
Nice job dude!

saledjenic commented:
@jrainville @borismelnik I added 3 commits more here:
-  updated a list of users who reacted with an emoji, since I've found that initially it was fulfilled by users' public key instead of name
- removed `NormalMessageView` component since we work with `CompactMessageView` and it's the only and default message component in the app, and removed `useCompactMode` property from local settings
- updated `ChannelIdentifierView` component

for point 3 we need to add only calls for `join group chat` and `leave group chat` from the chat identifier.
